### DISCUSSION TITLE
How to use Window.SetTitleBar() and have access to Caption Control Buttons?

### DISCUSSION THREAD

**Discussion by omnilogix** [microsoft/WindowsAppSDK#627] [Q&A]
I am following the example here:
https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.window.settitlebar?view=winui-3.0-preview

I can drag the window and right click to show the context menu.  The caption buttons are visible, but the entire area around the buttons is not responsive to clicking or dragging,

Is there a more complete example available that shows customizing the TitleBar and utilizing the caption buttons?

**Win32 Desktop**
![image](https://user-images.githubusercontent.com/9586533/112047634-cb334200-8b0a-11eb-921c-8d52140b3719.png)

**Comment by WelterDevelopment** [comment]
Set IsHitTestVisible="False" and do not set the background:
```xaml
<Grid  Grid.Row="0" Height="32" VerticalAlignment="Top" HorizontalAlignment="Stretch" x:Name="CustomDragRegion" IsHitTestVisible="False">
     <TextBlock VerticalAlignment="Center" HorizontalAlignment="Left" FontSize="20" Text="YOUR APP TITLE GOES HERE"></TextBlock>
</Grid>
```
Additionally, the new SetTitleBar()-Feature is a little 'dumb', e.g. its not possible to set the Titlebar to be the content of <TabView.TabStripFooter>, even if you set IsHitTestVisible="False", because the whole TabView itself is 'HitTestVisible' :(

**Reply by omnilogix** [reply]
@WelterDevelopment  Thanks for the tip, but it didn't work for me.  The buttons remain inactive and the content is also unresponsive with this solution.

**Reply by WelterDevelopment** [reply]
Steps to reproduce:
1. Create a new Project Reunion 0.5 App
2. Content of MainWindow.xaml:
```xaml
<Window
    x:Class="App6.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:App6"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d">

    <Grid >
        <Grid.RowDefinitions>
            <RowDefinition Height="Auto" />
            <RowDefinition Height="*" />
        </Grid.RowDefinitions>
        <Grid Grid.Row="0" Height="32"  x:Name="CustomDragRegion" IsHitTestVisible="False">
            <TextBlock VerticalAlignment="Center" HorizontalAlignment="Left" FontSize="20" Text="YOUR APP TITLE GOES HERE"></TextBlock>
        </Grid>
        <Grid Grid.Row="1" Background="DarkGray"></Grid>
    </Grid>
</Window>
```
3. Content of MainWindow.xaml.cs:
```csharp
using Microsoft.UI.Xaml;

namespace App6
{
    public sealed partial class MainWindow : Window
    {
        public MainWindow()
        {
            this.InitializeComponent();
            ExtendsContentIntoTitleBar = true;
            SetTitleBar(CustomDragRegion as UIElement);
        }
    }
}
```
4. Output:
![Scr_SetTitleBar](https://user-images.githubusercontent.com/13318246/114209096-89a1f400-995e-11eb-9801-7098fccc952c.png)

Does that work? If you have interactive content that is overlapping with your custom TitleBar you need a workaround; there is seemingly no solution to that.

**Reply by omnilogix** [reply]
@WelterDevelopment  Thanks for the detailed example!  I did get your example to work, but it falls short of my goal.  I think your final statement is what I'm dealing with.  I had intended to push the NavigationView up into the titlebar and let it's header be my new app title bar.   Unfortunately, the window buttons are behind the controls.  I think that I'm going to have to draw all of it custom, and PInvoke the window sizing myself.  At this point, it seem that the ExtendsContentIntoTitleBar is a big Fail.
